.github/workflows/libfuzzer.yaml

```diff
@@ -0,0 +1,170 @@
+name: Libfuzzer
+"on":
+  push:
+    branches:
+      - main
+      - prerelease_test
+  pull_request:
+    paths: .github/workflows/libfuzzer.yaml
+
+jobs:
+  fuzz:
+    name: Compression
+    runs-on: ubuntu-22.04
+    env:
+      PG_VERSION: 15.2
+      PG_SRC_DIR: pgbuild
+      PG_INSTALL_DIR: postgresql
+
+    steps:
+    - name: Install Linux Dependencies
+      run: |
+        # Don't add ddebs here because the ddebs mirror is always 503 Service Unavailable.
+        # If needed, install them before opening the core dump.
+        sudo apt-get update
+        sudo apt-get install clang lld llvm flex bison lcov systemd-coredump gdb libipc-run-perl \
+          libtest-most-perl
+
+    - name: Checkout TimescaleDB
+      uses: actions/checkout@v3
+
+    # We are going to rebuild Postgres daily, so that it doesn't suddenly break
+    # ages after the original problem.
+    - name: Get date for build caching
+      id: get-date
+      run: |
+        echo "date=$(date +"%d")" >> $GITHUB_OUTPUT
+    # we cache the build directory instead of the install directory here
+    # because extension installation will write files to install directory
+    # leading to a tainted cache
+    - name: Cache PostgreSQL
+      id: cache-postgresql
+      uses: actions/cache@v3
+      with:
+        path: ~/${{ env.PG_SRC_DIR }}
+        key: "postgresql-libfuzzer-${{ steps.get-date.outputs.date }}-${{ hashFiles('.github/**') }}"
+    - name: Build PostgreSQL
+      if: steps.cache-postgresql.outputs.cache-hit != 'true'
+      run: |
+       wget -q -O postgresql.tar.bz2 \
+         https://ftp.postgresql.org/pub/source/v$PG_VERSION/postgresql-$PG_VERSION.tar.bz2
+        mkdir -p ~/$PG_SRC_DIR
+        tar --extract --file postgresql.tar.bz2 --directory ~/$PG_SRC_DIR --strip-components 1
+        cd ~/$PG_SRC_DIR
+        CC=clang ./configure --prefix=$HOME/$PG_INSTALL_DIR --with-openssl \
+          --without-readline --without-zlib --without-libxml --enable-cassert \
+          --enable-debug CC=clang \
+          CFLAGS="-DTS_COMPRESSION_FUZZING=1 -fuse-ld=lld -ggdb3 -Og -fno-omit-frame-pointer"
+        make -j$(nproc)
+
+    - name: Install PostgreSQL
+      run: |
+        make -C ~/$PG_SRC_DIR install
+        make -C ~/$PG_SRC_DIR/contrib/postgres_fdw install
+
+    - name: Upload config.log
+      if: always()
+      uses: actions/upload-artifact@v3
+      with:
+        name: config.log for PostgreSQL
+        path: ~/${{ env.PG_SRC_DIR }}/config.log
+
+    - name: Build TimescaleDB
+      run: |
+        set -e
+
+        export LIBFUZZER_PATH=$(dirname "$(readlink -e "$(readlink -e $(llvm-config --libdir)/clang/*/lib/linux/libclang_rt.fuzzer_no_main-x86_64.a | head -1)")")
```

Comment:
A `find` in `$(llvm-config --libdir)` (as done for `"timescaledb-tsl-*.so` in Line 101) might shorten this long expression